gatherers/offline.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const Gather = require('./gather');
+
+// Request the current page by issuing a sync request to ''.
+const requestPage = `
+  (function() {
+    var request = new XMLHttpRequest();
+    request.open('GET', '', false);
+    request.send(null);
+    return request.status;
+  })();
+`;
+
+class Offline extends Gather {
+
+  static goOffline(driver) {
+    return driver.sendCommand('Network.emulateNetworkConditions', {
```

Comment:
add a comment here to indicate the `0` value means these are unset? (Because I'd typically expect `-1` for that, but hey devtools didn't abide)